atlp-rwanda/champs-ec-be
# #187300226 Configure ESLint and Git Pre commit hook to run tests and ESlint format

#### What does this PR do?

This PR implements the task related to configuring ESLint and Git pre-commit hooks to run tests and enforce ESLint formatting. It addresses the requirements outlined in Pivotal ID #187300226.
#### Description of Task to be completed?

Have the following working

• Set up ESLint for code linting and formatting

• Added ESLint configuration to enforce code style rules

• Configured Git pre-commit hook to run ESLint and tests before committing

• Added lint-staged to run ESLint on staged files before committing

#### How should this be manually tested?

1.Ensure ESLint is correctly configured by running the lint script (npm run lint) and verifying that code style rules are enforced.
2.Test the Git pre-commit hook by attempting to commit changes. Ensure that ESLint and tests are executed automatically before committing.
3.Verify that linting is applied to staged files by staging changes and then attempting to commit. Ensure that ESLint is run on staged files only.
What are the relevant pivotal tracker/Trello stories?

Pivotal ID #187300226.
